src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -0,0 +1,395 @@
+using System;
+using System.IO;
+using System.Text;
+using System.Security;
+using System.Management.Automation;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Management.Automation.Internal;
+// Once Serialization is available on CoreCLR: using System.Runtime.Serialization.Formatters.Binary;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// Displays the hexidecimal equivalent of the input data.
+    /// <summary>
+    [Cmdlet(VerbsCommon.Format, "Hex", SupportsShouldProcess = true, HelpUri ="https://go.microsoft.com/fwlink/?LinkId=821773")]
+    [Alias ("fhx")]
+    [OutputType(typeof(Microsoft.PowerShell.Commands.ByteCollection))]
+    public sealed class FormatHex : PSCmdlet
+    {
+
+        #region Internal Properties
+
+        private string[] _path;
+        private string[] _literalPath;
+        private PSObject _inputObject;
+        private string _encoding = "Ascii";
+        private const int BUFFERSIZE = 16;
+
+        #endregion
+
+        #region Parameters
+
+        /// <summary>
+        /// Path of file(s) to process
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0, ParameterSetName = "Path")]
+        [ValidateNotNullOrEmpty()]
+        public string[] Path 
+        { 
+            get { return _path; }
+            set { _path = value; }
+        }
+
+        /// <summary>
+        /// Literal path of file to process
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = "LiteralPath")]
+        [ValidateNotNullOrEmpty()]
+        [Alias("PSPath")]
+        public string[] LiteralPath
+        {
+            get { return _literalPath; }
+            set { _literalPath = value; }
+        }
+
+        /// <summary>
+        /// Ojbect to process
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = "ByInputObject", ValueFromPipeline = true)]
+        public PSObject InputObject
+        {
+            get { return _inputObject; }
+            set { _inputObject = value; }
+        }
+
+        /// <summary>
+        /// Type of character encoding for InputObject
+        /// </summary>
+        [Parameter(ParameterSetName = "ByInputObject")]
+        [ValidateSet("Ascii", "UTF32", "UTF7", "UTF8", "BigEndianUnicode", "Unicode")]
```

Comment:
yes, this should probably be an enum rather than the string set

---
In reply to: [105936094](https://github.com/PowerShell/PowerShell/pull/3320#discussion_r105936094) [](ancestors = 105936094)